sdk/tests/performance/webgl_webcodecs_video_frame/webgl_webcodecs_video_frame.js

```diff
@@ -0,0 +1,314 @@
+        gl.compileShader(shader);
+
+        if (!gl.getShaderParameter(shader, gl.COMPILE_STATUS)) {
+            console.log('An error occurred compiling the shaders: ' + gl.getShaderInfoLog(shader));
+            gl.deleteShader(shader);
+            return null;
+        }
+        return shader;
+    }
+
+    function initShaderProgram(gl, vsSource, fsSource) {
+        const vertexShader = loadShader(gl, gl.VERTEX_SHADER, vsSource);
+        const fragmentShader = loadShader(gl, gl.FRAGMENT_SHADER, fsSource);
+
+        const shaderProgram = gl.createProgram();
+        gl.attachShader(shaderProgram, vertexShader);
+        gl.attachShader(shaderProgram, fragmentShader);
+        gl.linkProgram(shaderProgram);
+
+        if (!gl.getProgramParameter(shaderProgram, gl.LINK_STATUS)) {
+            console.log('Unable to initialize the shader program: ' + gl.getProgramInfoLog(shaderProgram));
+            return null;
+        }
+
+        return shaderProgram;
+    }
+
+    // Computes a signature for the video frame to determine whether we need to
+    // re-compile GLSL shader to access it.
+    function generateSignature(videoFrameHandle) {
+        let signature = "";
+
+        signature += "{requiredExtension: " + videoFrameHandle.requiredExtension + "}\n";
+        signature += "{pixelFormat: " + videoFrameHandle.pixelFormat + "}\n";
+        let cs = videoFrameHandle.colorSpace;
+        signature += "{ColorSpace:\n  {";
+        signature += "primary: " + cs.primaryID + ",";
+        signature += "transfer: " + cs.transferID + ",";
+        signature += "matrix: " + cs.matrixID + ",";
+        signature += "range: " + cs.rangeID;
+        signature += "}\n";
+        signature += "}\n";
+
+        signature += "{flipY: " + videoFrameHandle.flipY + "}\n";
+        signature += "{preAlpha: " + videoFrameHandle.premultipliedAlpha + "}\n";
+
+        signature += "{\n";
+        let infoArray = videoFrameHandle.textureInfoArray;
+        for (let i = 0; i < infoArray.length; ++i) {
+            signature += "  { " + i.toString() + ": " + infoArray[i].samplerType
+                + ", " + infoArray[i].samplerFunc + ", "
+                + infoArray[i].components + "}\n";
+        }
+        signature += "}\n";
+        return signature;
+    }
+
+
+    function bindVideoFrame(gl, program, videoFrameHandle, texUnit, name) {
+        let infoArray = videoFrameHandle.textureInfoArray;
+        for (let i = 0; i < infoArray.length; ++i) {
+            gl.activeTexture(texUnit + i);
+            gl.bindTexture(infoArray[i].target, infoArray[i].texture);
+            gl.texParameteri(infoArray[i].target, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+            let uniformName = "_" + name + "_" + i.toString() + "_";
+            gl.uniform1i(gl.getUniformLocation(program, uniformName), i);
+        }
+    }
+
+    function initVertexBuffers(gl, program, videoFrame, posAttrib, coordAttrib) {
+        let vertices = new Float32Array([
```

Comment:
A lot of these "let"s could be "const"s since the variables are never modified afterward.

Reply:
Done.